### PR TITLE
Lang/Haven Barge

### PR DESCRIPTION
**Pull request by reithir**
Added support to bescort to enable taking the lang/riverhaven barge. Will couple it with some changes I will make to mapdb that will recruit 180 ranks in athletics to swim up the jantspyre (but no check to swim down). Here's my update script for mapdb for reference, feel free to run it when you merge branches or I can do it later:

Room[7633].timeto["7651"] = StringProc.new("unless running?('drinfomon') && UserVars.athletics >= 180 then nil else 0.2 end")
rh_dock = 50948
lang_dock = 3434
Room[lang_dock].timeto[rh_dock.to_s] = 600
Room[rh_dock].timeto[lang_dock.to_s] = 600
Room[lang_dock].wayto[rh_dock.to_s] = StringProc.new("start_script('bescort', ['lang_barge']);wait_while{running?('bescort')};")
Room[rh_dock].wayto[lang_dock.to_s] = StringProc.new("start_script('bescort', ['lang_barge']);wait_while{running?('bescort')};")

### CODE REVIEW

**Comment by rpherbig** [comment]
It looks like there is a conflict. Can you merge in master?

**Comment by reithir** [comment]
Uh oh, I'm not sure what it means to have a conflict; bescort.lic is the only file I edited. I'll click "resolve conflicts" and see what it wants me to do?

**Comment by rpherbig** [comment]
It means that you were working on an old version of the repo. You need to merge master back to your fork/branch.

You can use the GitHub web UI to fix the conflict and commit, or use the same workflow you used to make the first commit. Either way, any new commits will get automatically picked up by GitHub and added to the PR.

File: bescort.lic [hunk 1]
```diff
@@ -210,6 +213,8 @@ class Bescort
       take_xing_ferry(args.mode)
     elsif args.ferry1
       take_ain_ghazal_ferry(args.mode)
+	  elsif args.lang_barge
```
Comment (rpherbig) [inline bescort.lic]: It looks like you've got some tabs in here, could you adjust your editor to use spaces instead?

Another option would be to run Rubocop: `rubocop -a`

**Comment by reithir** [comment]
Ok I got it, I think my fixes should do the trick; ready to go. Thanks.

File: bescort.lic (hunk 1 above)
Comment (reithir) [reply bescort.lic]: Wow you're fast, sorry I'll update to spaces

File: bescort.lic [hunk 2]
```diff
@@ -1008,6 +1013,35 @@ class Bescort
...
+    end                                                                                                                                                              
+  when /The barge .* has just pulled away from the dock/, /The barge .* isn't docked, so you're stuck here until the next one arrives/                               
+    pause 1 while DRRoom.room_objs.find { |x| x =~ /the barge/ }                                                                                                     
+    take_rh_lang_barge(mode)                                                                                                                                         
+  when 'Come back when you can afford the fare'                                                                                                                      
+    echo('Your fare has mysteriously disappeared!')                                                                                                                  
+    return                                                                                                                                                           
+    end
```
Comment (Sarvatt) [inline bescort.lic]: 2 extra spaces? did you mean to have another end also to end take_rh_lang_barge?

File: bescort.lic [hunk 3]
```diff
@@ -1008,6 +1013,35 @@ class Bescort
...
+    if DRRoom.room_objs.find { |x| x =~ /Langenfirth wharf/ }                                                                                                        
+      move 'go wharf'                                                                                                                                                
+    else                                                                                                                                                             
+      move 'go pier'                                                                                                                                                 
+    end                                                                                                                                                              
+  when /The barge .* has just pulled away from the dock/, /The barge .* isn't docked, so you're stuck here until the next one arrives/                               
+    pause 1 while DRRoom.room_objs.find { |x| x =~ /the barge/ }                                                                                                     
+    take_rh_lang_barge(mode)
```
Comment (Sarvatt) [inline bescort.lic]: were you going to extend it to take the mode too?

File: bescort.lic [hunk 4]
```diff
@@ -1008,6 +1013,35 @@ class Bescort
...
+    end                                                                                                                                                              
+  when /The barge .* has just pulled away from the dock/, /The barge .* isn't docked, so you're stuck here until the next one arrives/                               
+    pause 1 while DRRoom.room_objs.find { |x| x =~ /the barge/ }                                                                                                     
+    take_rh_lang_barge                                                                                                                                         
+  when 'Come back when you can afford the fare'                                                                                                                      
+    echo('Your fare has mysteriously disappeared!')                                                                                                                  
+    return                                                                                                                                                           
+    end
```
Comment (jandersson) [inline bescort.lic]: You need one more end below this to close the method

Comment (reithir) [reply bescort.lic]: Fixed, thanks

File: bescort.lic [hunk 5]
```diff
@@ -1008,6 +1013,36 @@ class Bescort
       return
     end
   end
+
+def take_rh_lang_barge                                                                                                                                               
+  unless [50_948, 3434].include?(Room.current.id)
```
Comment (jandersson) [inline bescort.lic]: Can you add room 466 to this list? There are some mapping issues and 50948 is a duplicate of 466.

**Comment by jandersson** [comment]
The trip time is 557 seconds, for reference when updating `timeto` in the mapdb

File: bescort.lic (hunk 5 above)
Comment (reithir) [reply bescort.lic]: yeah I noticed that too, sometimes it's room 50_948 and other times it's 466. I'll just add 466 to the unless list, thanks.

**Comment by reithir** [comment]
Thanks for the time check; 600 wasn't a bad estimate!

**Comment by jandersson** [comment]
@rpherbig Ive ran this a few times. It works fine.

**Comment by reithir** [comment]
Resolved a conflict that came up, are we ready to merge this sucker into the master? thanks!

**Pull request merged**